docs/src/interfacer.md

```diff
@@ -0,0 +1,56 @@
+# Interfacer
+
+This module contains functions for defining the interface for coupling component models, as well as stub objects that contain prescribed fields.
+
+## Coupled Simulation
+- TODO: move `CoupledSimulation` from Utilities in a separate PR
+- `CoupledSimulation` (`cs`) stores info for ESM run. We require that each `cs` contains an `atmos_sim`, `ocean_sim` and `ice_sim` component. While this requirement will not be eventually needed, for the time being, if a simulation type is not needed, it should be initialized with `SurfaceStub` with a zero `area_fracion`.
+
+## Component model simulations
+- all Simulations that are not the `CoupledSimulation` fall under `ComponentModelSimulation`
+- the current version requires that there is:
+    - one `AtmosModelSimulation`
+    - one or more `SurfaceModelSimulation`s, which require the following adapter functions:
+        ```
+        get_field(sim::SurfaceModelSimulation, ::Val{:area_fraction}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:air_temperature}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:albedo}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:z0m}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:z0b}) = ...
+        get_field(sim::SurfaceModelSimulation, ::Val{:beta}) = ...
+        update_field!(sim::SurfaceModelSimulation, ::Val{:area_fraction}, field::Fields.Field) = ...
+        update_field!(sim::SurfaceModelSimulation, ::Val{:air_temperature}, field::Fields.Field) = ...
```

Comment:
Once we move the init functions to the component model repos, then yes, this will be a requirement. This doesn't mean they have to operate on ClimaCore spaces, but they need to be able to read the Field objects passed from the coupler (and so they'll depend on ClimaCore). We may want to generalize this in ClimaEarth to some universal objects, but for the AMIP setup everything depends on ClimaCore anyway.